common/persistence/sql/sqlExecutionManager.go

```diff
@@ -907,37 +908,49 @@ func (m *sqlExecutionManager) GetReplicationTasks(
 	request *p.GetReplicationTasksRequest,
 ) (*p.GetReplicationTasksResponse, error) {
 
-	var readLevel int64
-	var maxReadLevelInclusive int64
-	var err error
-	if len(request.NextPageToken) > 0 {
-		readLevel, err = deserializePageToken(request.NextPageToken)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		readLevel = request.ReadLevel
+	readLevel, maxReadLevelInclusive, err := getReadLevels(request)
+	if err != nil {
+		return nil, err
 	}
-	maxReadLevelInclusive = collection.MaxInt64(
-		readLevel+int64(request.BatchSize), request.MaxReadLevel)
 
-	rows, err := m.db.SelectFromReplicationTasks(&sqldb.ReplicationTasksFilter{
-		ShardID:   m.shardID,
-		MinTaskID: &readLevel,
-		MaxTaskID: &maxReadLevelInclusive,
-		PageSize:  &request.BatchSize,
-	})
-	if err != nil {
-		if err != sql.ErrNoRows {
-			return nil, &workflow.InternalServiceError{
-				Message: fmt.Sprintf("GetReplicationTasks operation failed. Select failed: %v", err),
-			}
+	rows, err := m.db.SelectFromReplicationTasks(
+		&sqldb.ReplicationTasksFilter{
+			ShardID:   m.shardID,
+			MinTaskID: readLevel,
+			MaxTaskID: maxReadLevelInclusive,
+			PageSize:  request.BatchSize,
+		})
+
+	if err != nil && err != sql.ErrNoRows {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetReplicationTasks operation failed. Select failed: %v", err),
 		}
 	}
```

Comment:
a switch case may make this looks cleaner

Reply:
done